packages/clay-data-provider/src/LRUCache.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const DEFAULT_MAX_SIZE = 20;
+
+export class LRUCache<K, V> {
+	maxSize: number;
+	storage: Map<K, V>;
```

Comment:
Shouldn't these be `private` or at least `_maxSize`/`_storage`? Otherwise I can break encapsulation.

Reply:
Oh yeah, I'm hoping no one will touch the instance of `LRUCache`, I tried to use `Symbol` as a key in `useCache` to avoid these kinds of things. But I was thinking of going with `_maxSize`/`_storage` but I find the underline so horrible for that 😅, I will go with `private` for more security here.

Reply:
Yeah, seeing as this is a TypeScript project, `private` (and `#` private class fields once those get implemented) is probably preferable over `_` prefix. Even with `private` it is not literally private, it is only advisory. TypeScript will tell you if you do it wrong, but you can cast away:

```javascript
// Property 'x' is private and only accessible within class 'Thing'.
thing.x = 11;

// 😃 
(thing as any).x = 11;
```

But still, its good to put some measure in place to stop me going `cache.storage.clear()` from the outside.